rust/processor/src/models/ans_models/ans_lookup.rs

```diff
@@ -0,0 +1,776 @@
+use field_count::FieldCount;
+use serde::{Deserialize, Serialize};
+use std::{cmp, collections::HashMap};
+
+pub const QUERY_RETRIES: u32 = 5;
+pub const QUERY_RETRY_DELAY_MS: u64 = 500;
+
+type Domain = String;
+type Subdomain = String;
+// Hashmap for looking up previous name records by target address
+pub type AddressToPrimaryNameRecord = HashMap<String, CurrentAnsLookup>;
+// PK of current_ans_lookup, i.e. domain and subdomain name
+pub type CurrentAnsLookupPK = (Domain, Subdomain);
+// PK of ans_lookup, i.e. transaction version, domain, and subdomain
+pub type AnsLookupPK = (i64, Domain, Subdomain);
+
+#[derive(Clone, Default, Debug, Deserialize, FieldCount, Identifiable, Insertable, Serialize)]
+#[diesel(primary_key(domain, subdomain))]
+#[diesel(table_name = current_ans_lookup)]
+#[diesel(treat_none_as_null = true)]
+pub struct CurrentAnsLookup {
+    pub domain: String,
+    pub subdomain: String,
+    pub registered_address: Option<String>,
+    pub last_transaction_version: i64,
+    pub expiration_timestamp: chrono::NaiveDateTime,
+    pub token_name: String,
+    pub is_primary: bool,
+    pub is_deleted: bool,
+}
+
+/// Need a separate struct for queryable because we don't want to define the inserted_at column (letting DB fill)
+#[derive(Debug, Identifiable, Queryable)]
+#[diesel(primary_key(domain, subdomain))]
+#[diesel(table_name = current_ans_lookup)]
+pub struct CurrentAnsLookupDataQuery {
+    pub domain: String,
+    pub subdomain: String,
+    pub registered_address: Option<String>,
+    pub expiration_timestamp: chrono::NaiveDateTime,
+    pub last_transaction_version: i64,
+    pub inserted_at: chrono::NaiveDateTime,
+    pub token_name: String,
+    pub is_primary: bool,
+    pub is_deleted: bool,
+}
+
+#[derive(Clone, Default, Debug, Deserialize, FieldCount, Identifiable, Insertable, Serialize)]
+#[diesel(primary_key(transaction_version, domain, subdomain))]
+#[diesel(table_name = ans_lookup)]
+#[diesel(treat_none_as_null = true)]
+pub struct AnsLookup {
+    pub transaction_version: i64,
+    pub domain: String,
+    pub subdomain: String,
+    pub registered_address: Option<String>,
+    pub expiration_timestamp: chrono::NaiveDateTime,
+    pub token_name: String,
+    pub is_primary: bool,
+    pub is_deleted: bool,
+    pub transaction_timestamp: chrono::NaiveDateTime,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+struct OptionalString {
+    vec: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct NameRecordKeyV1 {
+    domain_name: String,
+    subdomain_name: OptionalString,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct NameRecordV1 {
+    #[serde(deserialize_with = "deserialize_from_string")]
+    expiration_time_sec: BigDecimal,
+    #[serde(deserialize_with = "deserialize_from_string")]
+    property_version: BigDecimal,
+    target_address: OptionalString,
+}
+
+impl CurrentAnsLookup {
+    pub fn from_transaction(
```

Comment:
If you're going to make a new processor, why not just move this logic to the processor? It'll be better b/c when you add v2 you don't need to make multiple loops again in the transaction